ui/button.js

```diff
@@ -56,355 +39,267 @@ $.widget( "ui.button", {
 	defaultElement: "<button>",
 	options: {
 		disabled: null,
-		text: true,
+		showLabel: true,
 		label: null,
-		icons: {
-			primary: null,
-			secondary: null
+		icon: null,
+		iconPosition: "beginning",
+		classes: {
+			"ui-button": "ui-corner-all"
 		}
 	},
-	_create: function() {
-		this.element.closest( "form" )
-			.unbind( "reset" + this.eventNamespace )
-			.bind( "reset" + this.eventNamespace, formResetHandler );
 
-		if ( typeof this.options.disabled !== "boolean" ) {
-			this.options.disabled = !!this.element.prop( "disabled" );
-		} else {
-			this.element.prop( "disabled", this.options.disabled );
-		}
+	_getCreateOptions: function() {
+		var disabled,
+			options =  this._super() || {};
 
-		this._determineButtonType();
-		this.hasTitle = !!this.buttonElement.attr( "title" );
+		this.isInput = this.element.is( "input" );
+		this.originalLabel = this.isInput ? this.element.val() : this.element.html();
 
-		var that = this,
-			options = this.options,
-			toggleButton = this.type === "checkbox" || this.type === "radio",
-			activeClass = !toggleButton ? "ui-state-active" : "";
+		disabled = this.element[ 0 ].disabled;
+		if ( disabled != null ) {
+			options.disabled = disabled;
+		}
 
-		if ( options.label === null ) {
-			options.label = (this.type === "input" ? this.buttonElement.val() : this.buttonElement.html());
+		if ( this.originalLabel ) {
+			options.label = this.originalLabel;
 		}
 
-		this._hoverable( this.buttonElement );
+		return options;
+	},
 
-		this.buttonElement
-			.addClass( baseClasses )
-			.attr( "role", "button" )
-			.bind( "mouseenter" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				if ( this === lastActive ) {
-					$( this ).addClass( "ui-state-active" );
-				}
-			})
-			.bind( "mouseleave" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				$( this ).removeClass( activeClass );
-			})
-			.bind( "click" + this.eventNamespace, function( event ) {
-				if ( options.disabled ) {
-					event.preventDefault();
-					event.stopImmediatePropagation();
-				}
-			});
+	_create: function() {
+		var formElement = $( this.element[ 0 ].form );
 
-		// Can't use _focusable() because the element that receives focus
-		// and the element that gets the ui-state-focus class are different
-		this._on({
-			focus: function() {
-				this.buttonElement.addClass( "ui-state-focus" );
-			},
-			blur: function() {
-				this.buttonElement.removeClass( "ui-state-focus" );
-			}
-		});
+		// We don't use _on and _off here because we want all the checkboxes in the same form to use
+		// single handler which handles all the checkboxradio widgets in the form
+		formElement.off( "reset" + this.eventNamespace, formResetHandler );
+		formElement.on( "reset" + this.eventNamespace, formResetHandler );
```

Comment:
This also isn't being removed on destroy. Neither of these problems are new though.

We have a somewhat similar situation in dialog where we need to do cross-instance behavior management. Here's the general idea for how to fix this in button:

```js
if ( !form.data( "ui-button-reset" ) ) {
    form
        .on( "reset.ui-button", formResetHandler )
        .data( "ui-button-reset", true );
}
```

In `_destroy()` you then check if there are any buttons left. Another option is to track the number of buttons in `_create()` and decrement in `_destroy()` so that you can then check the count rather than searching the DOM.